### DISCUSSION TITLE
add my project  to template format

### DISCUSSION THREAD

**Discussion by myeghaneh** [explosion/spaCy#7558] [Help: Coding & Implementations]
I want to add my project which is a custom NER (with standard tagger and dep of sm model), in this project, I have provided 

dev.spacy
train.spacy 
 

I have followed this  template 

https://github.com/explosion/projects

tutorials/ner_fashion_brands

but as you here, this template is kind of starting from the zero, 

since I have my data, as well as model, how can I use a template for my case?

**Comment by myeghaneh** [comment]
I have kind of solved this using  

https://github.com/explosion/projects/tree/v3/pipelines/ner_demo_update

ner_demo_update

now work great if you give them  a  dev,test binary file (.spacy) then it provides the mode and the rest...


but my question is how can I give the json (or jsonl) file then the result is as  ner_demo_update


since I have annotated my data and it is in the correct format json like this...
[
["I have presented these models hypothetically, the hypothesis being astronomy's testimony that the planet's path is a perfect eccentric circle such as was described. If astronomy will discover something different, the physical theories will also change.", {"entities": [[125, 134, "MODELL"], [79, 88, "NAME"]]}], 
...]


how can I make an end to end template, giving JSON flies (dev,train) it provided the binary file and as well as models 

should I use a different template?

**Reply by adrianeboyd** [reply]
I'm not sure I understand the question? The sample data in that project is in the same format as this, which you can generate from v2 examples with `srsly.write_json("file.json", TRAIN_DATA)`. Or if you have a different format, all you need to do is customize is the convert commands/scripts.

**Reply by myeghaneh** [reply]
thank you for the quick response. 

my data  is   JSON files  with this format : 

[
["I have presented these models hypothetically, the hypothesis being astronomy's testimony that the planet's path is a perfect eccentric circle such as was described. If astronomy will discover something different, the physical theories will also change.", {"entities": [[125, 134, "MODELL"], [79, 88, "NAME"]]}],
...]



my question is which template should I use to have an end-to-end pipeline.. I could manage to use this template (ner_demo_update)

https://github.com/explosion/projects/tree/v3/pipelines/ner_demo_update

 for my binary file (.spacy)  that I provided by my script from that JSON files. 

 but I want to have a real end-to-end pipeline that only required the JSON file (annotation) and gives the model and the result by those commands. 

which template should I use for this aim?  

I hope I explained it properly now. 


Many thanks

**Comment by myeghaneh** [comment]
I would be happy if anyone can help , many thanks in advance
